pymc3/distributions/continuous.py

```diff
@@ -634,8 +634,13 @@ def logp(self, value):
         alpha = self.alpha
         beta = self.beta
 
-        return bound(logpow(value, alpha - 1) + logpow(1 - value, beta - 1)
-                     - betaln(alpha, beta),
+        logval = tt.log(value)
+        log1pval = tt.log1p(-value)
+        logp = (tt.switch(tt.eq(alpha, 1), 0, (alpha - 1) * logval)
```

Comment:
Hi @aseyboldt, I have a question for you. Without `tt.switch` results should be the same, so why it is necessary? Are  results somehow more stable?

Reply:
They aren't the same because `0. * (-inf) == nan`, not 0.
I'm not entirely sure if we really should worry about this here, because value == 0 is only on a null-set, but there was a bug report that the value was wrong.

Reply:
ohhh I see, thanks for clarification.